### PR TITLE
cherry pick of #9266

### PR DESCRIPTION
**Pull request by dimapihtar**
# What does this PR do ?

cherry pick of #9266 PR.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: .github/workflows/cicd-main.yml [hunk 1]
```diff
@@ -89,10 +89,91 @@ jobs:
...
+
+    # - name: Inspect
+    #   run: |
+    #     docker buildx imagetools inspect nemoci.azurecr.io/name/app:latest
+
+    #- name: Post-workflow execution
+    #  uses: gacts/run-and-post-run@v1
+    #  with:
+    #    post: |
+    #      chmod -R 777 .
```
Comment (ericharper) [inline .github/workflows/cicd-main.yml:176]: I think we can revert this, just remove the tests that are not needed

File: .github/workflows/cicd-main.yml [hunk 2]
```diff
@@ -1655,159 +1214,70 @@ jobs:
...
+              model.test_ds.ds_item="${data_dir}" \
+              pretrained_model="${output_dir}/checkpoints/Punctuation_and_Capitalization.nemo" \
+              +model.train_ds.use_cache=false \
+              +model.validation_ds.use_cache=false \
+              +model.test_ds.use_cache=false \
+              trainer.devices=[0,1] \
+              trainer.strategy=ddp \
+              trainer.max_epochs=1 \
+              exp_manager=null && \
+            rm -rf "${work_dir}"
```
Comment (ericharper) [inline .github/workflows/cicd-main.yml:1280]: I don't think we want this

File: .github/workflows/cicd-main.yml [hunk 3]
```diff
@@ -89,11 +89,37 @@ jobs:
...
+    # - name: Inspect
+    #   run: |
+    #     docker buildx imagetools inspect nemoci.azurecr.io/name/app:latest
+
+    #- name: Post-workflow execution
+    #  uses: gacts/run-and-post-run@v1
+    #  with:
+    #    post: |
+    #      chmod -R 777 .
+
```
Comment (ericharper) [inline .github/workflows/cicd-main.yml:122]: I don't think we want these changes?

File: .github/workflows/cicd-main.yml [hunk 4]
```diff
@@ -89,11 +89,37 @@ jobs:
           python setup.py style
           python tests/check_copyright_header.py --dir .
 
-          # These checks are not crucial
-          exit 0
+        # These checks are not crucial
+        exit 0
```
Comment (ericharper) [inline .github/workflows/cicd-main.yml:93]: why change this?

File: .github/workflows/cicd-main.yml (hunk 3 above)
Comment (dimapihtar) [reply .github/workflows/cicd-main.yml]: resolved

File: .github/workflows/cicd-main.yml (hunk 4 above)
Comment (dimapihtar) [reply .github/workflows/cicd-main.yml]: @ericharper just a typo
resolved

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**